src/com/trollworks/gcs/skill/SkillOutline.java

```diff
@@ -157,6 +158,71 @@ public void increment() {
 	}
 
 	@Override
+	public boolean canIncrementSkillLevel() {
+		return canIncrement();
+	}
+
+	@Override
+	public boolean canDecrementSkillLevel() {
+		return canDecrement();
+	}
+
+	@SuppressWarnings("unused")
+	@Override
+	public void incrementSkillLevel() {
+		List<RowUndo> undos = new ArrayList<>();
+		for (Skill skill : new FilteredIterator<>(getModel().getSelectionAsList(), Skill.class)) {
+			if (!skill.canHaveChildren()) {
+				int basePoints = skill.getPoints() + 1;
+				int maxPoints = basePoints + (skill.getDifficulty() == SkillDifficulty.W ? 12 : 4);
```

Comment:
The idea was to prevent infinite loop in case if level is never increased (for techniques with upper limit). And now I realized I forgot to handle this situation, I should reset points to previous value in that case.